base_layer/wallet/src/key_manager_service/storage/sqlite_db/mod.rs

```diff
@@ -135,13 +131,13 @@ impl KeyManagerBackend for KeyManagerSqliteDatabase {
         let start = Instant::now();
         let conn = self.database_connection.get_pooled_connection()?;
         let acquire_lock = start.elapsed();
-        let mut km = KeyManagerStateSql::get_state(&branch, &conn)?;
-        self.decrypt_if_necessary(&mut km)?;
+        let km = KeyManagerStateSql::get_state(&branch, &conn)?;
+        let mut km = self.decrypt_value(km)?;
```

Comment:
This probably doesn't need to be encrypted, the key index can be changed without decrypting

Reply:
I removed the decryption for `KeyManagerStateSql`.